analyzere/base_resources.py

```diff
@@ -229,10 +229,17 @@ def upload_data(self, file_or_str):
         """
         file_obj = StringIO(file_or_str) if isinstance(
             file_or_str, six.string_types) else file_or_str
-        length = utils.file_length(file_obj)
 
-        # Initiate upload session
-        request_raw('post', self._data_path, headers={'Entity-Length': length})
+        # Upload file with known entity size if file object supports random
+        # access.
+        if hasattr(file_obj, 'seek'):
```

Comment:
The correct way to test for random access should be the [`seekable()`](https://docs.python.org/3/library/io.html#io.IOBase.seekable) method. Oddly enough `six` does not implement that for py27 `StringIO`, even though py3.5 implements it.

@pmdarrow What would be your suggestion on how to deal with that? I was thinking to test if `seekable` was implemented and if not or `False` assume no random access, which means `StringIO` on py27 will be treated as a stream and while on py35 as a random access file. Alternatively, for our internal use of `StringIO` in `base_resources.py` and the tests I can patch the py27 version to implement `seekable()`.